core/task/manager.go

```diff
@@ -783,9 +783,15 @@ func (m *Manager) handleMessage(tm *TaskmanMessage) (error) {
 	case taskop.AcquireTasks:
 		go m.acquireTasks(tm.GetEnvironmentId(), tm.GetDescriptors())
```

Comment:
Now that I look at it again I'm a bit concerned about this (& apologies if I'm reopening a previously settled matter, I'd just like to be doubly sure). `acquireTasks` is not mutexed from other operations, yet the fact that all of them here are dispatched as separate goroutines makes it possible for e.g. `acquireTasks` to start concurrently with `transitionTasks`.

Is the thread safety of `m.roster` enough to guarantee the overall thread safety of these concurrent calls?

Reply:
The `m.roster` is safe when it comes to doing operations with the roster. But with the current `transition_configure.go` it is impossible to make a `transitionTasks` call before the `acquireTasks` is finished. Cause in the `transition_configure.go` we wait for the `wfStatus` to be active, before issuing a `configureTasks` call.